lib/ui/dart_ui.gni

```diff
@@ -2,26 +2,55 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-dart_ui_files = [
-  "//flutter/lib/ui/annotations.dart",
-  "//flutter/lib/ui/channel_buffers.dart",
-  "//flutter/lib/ui/compositing.dart",
-  "//flutter/lib/ui/geometry.dart",
-  "//flutter/lib/ui/hash_codes.dart",
-  "//flutter/lib/ui/hooks.dart",
-  "//flutter/lib/ui/isolate_name_server.dart",
-  "//flutter/lib/ui/key.dart",
-  "//flutter/lib/ui/lerp.dart",
-  "//flutter/lib/ui/math.dart",
-  "//flutter/lib/ui/natives.dart",
-  "//flutter/lib/ui/painting.dart",
-  "//flutter/lib/ui/platform_dispatcher.dart",
-  "//flutter/lib/ui/plugins.dart",
-  "//flutter/lib/ui/pointer.dart",
-  "//flutter/lib/ui/semantics.dart",
-  "//flutter/lib/ui/text.dart",
-  "//flutter/lib/ui/ui.dart",
-  "//flutter/lib/ui/window.dart",
-]
+import("//flutter/impeller/tools/impeller.gni")
 
-dart_ui_path = "//flutter/lib/ui/ui.dart"
+if (!impeller_enable_3d) {
```

Comment:
You can factor out the common lists and use `+` to add the ones that are different.
```gn
_common_dart_ui_files = [
  ...
]

if () {
  dart_ui_files = _common_dart_ui_files + [
  ]
} else {
}
```

Also, think positive:
```gn
if (impeller_enable_3d) {
} else {
}
```